src/main/java/com/epam/grid/engine/provider/utils/DirectoryPathUtils.java

```
@@ -0,0 +1,111 @@
+                        log.error(errorMessage);
+                        throw new IOException(errorMessage);
+                    }
+                } catch (final Exception exception) {
+                    log.error(exception.getMessage());
+                    exception.printStackTrace();
+                }
+            }
+            properDir.append(adjustableFolderPath);
+        } else {
+            if (gridEngineFolder.endsWith(FORWARDSLASH)) {
+                properDir.append(gridEngineFolder).append(adjustableFolderPath);
+            } else {
+                properDir.append(gridEngineFolder).append(FORWARDSLASH).append(adjustableFolderPath);
+            }
+            log.info("Provided directory was changed to " + properDir);
+            final File gridEnginePath = new File(properDir.toString());
+            if (!gridEnginePath.exists()) {
+                try {
+                    if (gridEnginePath.mkdirs()) {
+                        log.info(gridEnginePath + " Grid Engine folder was created.");
+                        grantAllPermissionsToFolder(gridEnginePath);
+                    } else {
+                        final String errorMessage = "Failed to create directory with path " + gridEnginePath;
+                        log.error(errorMessage);
+                        throw new IOException(errorMessage);
+                    }
+                } catch (final Exception exception) {
+                    log.error(exception.getMessage());
+                    exception.printStackTrace();
+                }
+            }
+        }
+        return properDir.toString();
+    }
+
+    private static void grantAllPermissionsToFolder(final File directory) {
+        try {
+            final Set<PosixFilePermission> permissions = PosixFilePermissions.fromString(ALL_PERMISSIONS_STRING);
+            Files.setPosixFilePermissions(directory.toPath(), permissions);
+        } catch (final IOException exception) {
```

here you swallow and exception and this is not good, please refer to best practice https://dzone.com/articles/9-best-practices-to-handle-exceptions-in-java (7, 8, 9)

---

fixed

---

So now, we catch a checked exception and somehow handle it (by throw an unchecked one), good, but actually I would handle IOException not here but in higher level ( in `buildProperDir`).
Main rule for such cases is that:
 - low level methods like `grantAllPermissionsToFolder` shouldn't decide what to do with exceptional situation but rather they should just inform about it (so throws a checked exception)
 - method with higher level should deal with handling checked exception, because they "know" business logic and "understand" what they need to do (throw unchecked exception and fail or for example retry)

---

done